local/network_test.go

```diff
@@ -612,6 +612,86 @@ func TestGetAllNodes(t *testing.T) {
 	}
 }
 
+// TestFlags tests that we can pass flags through the network.Config
+// but also via node.Config and that the latter overrides the former
+// if same keys exist.
+func TestFlags(t *testing.T) {
+	assert := assert.New(t)
+	networkConfig := testNetworkConfig(t)
+
+	// submit both network.Config flags and node.Config
+	networkConfig.Flags = map[string]interface{}{
+		"test-network-config-flag": "something",
+		"common-config-flag":       "should not be added",
+	}
+	for i := range networkConfig.NodeConfigs {
+		v := &networkConfig.NodeConfigs[i]
+		v.Flags = map[string]interface{}{
+			"test-node-config-flag":  "node",
+			"test2-node-config-flag": "config",
+			"common-config-flag":     "this should be added",
+		}
+	}
+	nw, err := newNetwork(logging.NoLog{}, networkConfig, newMockAPISuccessful, newMockProcessSuccessful)
+	assert.NoError(err)
+	// after creating the network, one flag should have been overridden by the node configs
+	for _, n := range networkConfig.NodeConfigs {
+		assert.True(len(n.Flags) == 4)
+		assert.Contains(n.Flags, "test-network-config-flag")
```

Comment:
probably use assert.Equal for all flags and check all expected values in all cases? no sure it is checking that network config flags are beeing copied